HopsanGUI/Widgets/TextEditorWidget.cpp

```diff
@@ -112,6 +131,16 @@ void TextEditorWidget::find(QString text, QTextDocument::FindFlags flags)
     mpEditor->find(text,flags);
 }
 
+void TextEditorWidget::fileChanged(QString filePath)
+{
+    qDebug() << "File changed: " << filePath;
+    //We need to add the path again, because some text editors removes the actual file and
+    //creates a new one when saving (e.g. gedit). This makes the watcher stop watching.
+    qobject_cast<QFileSystemWatcher*>(sender())->addPath(filePath);
```

Comment:
While tight now, sender is always a QFileSystemWatcher, there should be a pointer check on the result of qobject_cast, just in case someone else sends the signal, or the code will crash due to nullptr.

Reply:
That's true. This is actually a problem on many places already. It works for now, but only until someone changes something sometimes in the future...